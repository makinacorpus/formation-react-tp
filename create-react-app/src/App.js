import React from 'react';
import HelloWorld from './components/HelloWorld/';
import NominatimResults from './components/NominatimResults/';
import OverpassResults from './components/OverpassResults/';

import './App.css';

class App extends React.Component {
  render() {
    return (
      <div className="App">
        <HelloWorld name="World !" />
        NominatimResults
        <NominatimResults />
        OverpassResults
        <OverpassResults />
      </div>
    );
  }
}

export default App;
