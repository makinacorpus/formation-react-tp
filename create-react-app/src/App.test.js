import React from 'react';
import ReactDOM from 'react-dom';

import App from './App';

import axios from 'axios' // v0.15.3
import httpAdapter from 'axios/lib/adapters/http'
import nock from 'nock' // v9.0.2

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const host = 'http://localhost';

axios.defaults.host = host;
axios.defaults.adapter = httpAdapter;

describe('<App /> component', () => {
  nock(host)
    .get('*')
    .reply(200, 'test data');

  const wrapper = shallow(<App />);

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an .App element', () => {
    expect(wrapper.find('.App')).toHaveLength(1);
  });

  it('should render a Nominatim component', () => {
    expect(wrapper.find('Nominatim')).toHaveLength(1);
  });

  it('should render a Map component', () => {
    expect(wrapper.find('Map')).toHaveLength(1);
  });
});
