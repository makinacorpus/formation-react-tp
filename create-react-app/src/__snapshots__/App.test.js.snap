// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nominatim
          data={Array []}
          handleChange={[Function]}
          handleInput={[Function]}
          handleSubmitControlled={[Function]}
          handleSubmitUncontrolled={[Function]}
          inputValue="Toulouse"
/>,
        <Map
          changeBBox={[Function]}
          geojson={undefined}
          id="map"
          markers={Array []}
/>,
        <RefreshIndicator
          left={70}
          loadingColor="#FF9800"
          percentage={0}
          size={50}
          status="hide"
          style={
                    Object {
                              "display": "inline-block",
                              "left": "unset",
                              "position": "absolute",
                              "right": "1rem",
                              "top": "1rem",
                              "transform": "unset",
                              "zIndex": 10000,
                            }
          }
          top={0}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Array [],
          "handleChange": [Function],
          "handleInput": [Function],
          "handleSubmitControlled": [Function],
          "handleSubmitUncontrolled": [Function],
          "inputValue": "Toulouse",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "changeBBox": [Function],
          "geojson": undefined,
          "id": "map",
          "markers": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "left": 70,
          "loadingColor": "#FF9800",
          "percentage": 0,
          "size": 50,
          "status": "hide",
          "style": Object {
            "display": "inline-block",
            "left": "unset",
            "position": "absolute",
            "right": "1rem",
            "top": "1rem",
            "transform": "unset",
            "zIndex": 10000,
          },
          "top": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nominatim
            data={Array []}
            handleChange={[Function]}
            handleInput={[Function]}
            handleSubmitControlled={[Function]}
            handleSubmitUncontrolled={[Function]}
            inputValue="Toulouse"
/>,
          <Map
            changeBBox={[Function]}
            geojson={undefined}
            id="map"
            markers={Array []}
/>,
          <RefreshIndicator
            left={70}
            loadingColor="#FF9800"
            percentage={0}
            size={50}
            status="hide"
            style={
                        Object {
                                    "display": "inline-block",
                                    "left": "unset",
                                    "position": "absolute",
                                    "right": "1rem",
                                    "top": "1rem",
                                    "transform": "unset",
                                    "zIndex": 10000,
                                  }
            }
            top={0}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [],
            "handleChange": [Function],
            "handleInput": [Function],
            "handleSubmitControlled": [Function],
            "handleSubmitUncontrolled": [Function],
            "inputValue": "Toulouse",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "changeBBox": [Function],
            "geojson": undefined,
            "id": "map",
            "markers": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "left": 70,
            "loadingColor": "#FF9800",
            "percentage": 0,
            "size": 50,
            "status": "hide",
            "style": Object {
              "display": "inline-block",
              "left": "unset",
              "position": "absolute",
              "right": "1rem",
              "top": "1rem",
              "transform": "unset",
              "zIndex": 10000,
            },
            "top": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
